scale-tests/configs/2019-02-28-mwst13.env

```diff
@@ -0,0 +1,147 @@
+
+# Common streaming jobs configuration ##########################################
+
+TEST_ASSEMBLY_JAR_URL='http://infinity-artifacts.s3.amazonaws.com/scale-tests/dcos-spark-scala-tests-assembly-20180612-5fa9420.jar'
+NUM_DISPATCHERS="$((${NON_GPU_NUM_DISPATCHERS} + ${GPU_NUM_DISPATCHERS}))"
+DISPATCHERS_JSON_OUTPUT_FILE="all-dispatchers.json"
+
+# Failing streaming jobs configuration #########################################
+
+FAILING_SUBMISSIONS_OUTPUT_FILE="failing-submissions.out"
+FAILING_NUM_PRODUCERS_PER_KAFKA="${NON_GPU_NUM_DISPATCHERS}"
+FAILING_NUM_CONSUMERS_PER_PRODUCER=1
+FAILING_PRODUCER_NUMBER_OF_WORDS=7692
+FAILING_PRODUCER_WORDS_PER_SECOND=1
+FAILING_PRODUCER_SPARK_CORES_MAX=2
+FAILING_PRODUCER_SPARK_EXECUTOR_CORES=2
+FAILING_CONSUMER_BATCH_SIZE_SECONDS=10
+FAILING_CONSUMER_SPARK_CORES_MAX=1
+FAILING_CONSUMER_SPARK_EXECUTOR_CORES=1
+
+# Finite streaming jobs configuration ##########################################
+
+FINITE_SUBMISSIONS_OUTPUT_FILE="finite-submissions.out"
+FINITE_NUM_PRODUCERS_PER_KAFKA="${NON_GPU_NUM_DISPATCHERS}" # 1 Kafka and 50 dispatchers -> 50 producers.
+FINITE_NUM_CONSUMERS_PER_PRODUCER=1 # 50 producers -> 50 consumers.
+# 50 producers + 50 consumers = 100 total finite streaming jobs
+FINITE_PRODUCER_NUMBER_OF_WORDS=7692
+FINITE_PRODUCER_WORDS_PER_SECOND=1
+# 7692 words / 1 word per second -> ~2h runtime.
+FINITE_PRODUCER_SPARK_CORES_MAX=2
+FINITE_PRODUCER_SPARK_EXECUTOR_CORES=2
+FINITE_CONSUMER_BATCH_SIZE_SECONDS=10
+FINITE_CONSUMER_SPARK_CORES_MAX=1
+FINITE_CONSUMER_SPARK_EXECUTOR_CORES=1
+
+# Infinite streaming jobs configuration ########################################
+
+INFINITE_SUBMISSIONS_OUTPUT_FILE="infinite-submissions.out"
+INFINITE_NUM_PRODUCERS_PER_KAFKA="${NON_GPU_NUM_DISPATCHERS}" # 1 Kafka and 50 dispatchers -> 50 producers.
+INFINITE_NUM_CONSUMERS_PER_PRODUCER=1 # 50 producers -> 50 consumers.
+# 50 producers + 50 consumers = 100 total infinite streaming jobs
+INFINITE_PRODUCER_NUMBER_OF_WORDS=0
+INFINITE_PRODUCER_WORDS_PER_SECOND=1
+INFINITE_PRODUCER_SPARK_CORES_MAX=2
+INFINITE_PRODUCER_SPARK_EXECUTOR_CORES=2
+INFINITE_CONSUMER_BATCH_SIZE_SECONDS=10
+INFINITE_CONSUMER_SPARK_CORES_MAX=1
+INFINITE_CONSUMER_SPARK_EXECUTOR_CORES=1
+
+# Batch jobs configuration #####################################################
+
+NON_GPU_DISPATCHERS_JSON_OUTPUT_FILE_URL="https://${TEST_S3_BUCKET}.s3.amazonaws.com/${TEST_S3_FOLDER}/${NON_GPU_DISPATCHERS_JSON_OUTPUT_FILE}"
+
+BATCH_APP_ID="/${SERVICE_NAMES_PREFIX}batch-workload"
+BATCH_SCRIPT_CPUS=6
+BATCH_SCRIPT_MEM=12288
+BATCH_SUBMITS_PER_MIN=13
+BATCH_SPARK_BUILD_BRANCH=master
+
+# Batch GPU jobs configuration #################################################
+
+GPU_DISPATCHERS_JSON_OUTPUT_FILE_URL="https://${TEST_S3_BUCKET}.s3.amazonaws.com/${TEST_S3_FOLDER}/${GPU_DISPATCHERS_JSON_OUTPUT_FILE}"
+
+GPU_APP_ID="/${SERVICE_NAMES_PREFIX}gpu-batch-workload"
+GPU_SCRIPT_CPUS=2
+GPU_SCRIPT_MEM=4096
+GPU_DOCKER_IMAGE='samvantran/spark-dcos-gpu:metrics'
```

Comment:
This image was built on spark 2.2.1 and given that we are now soaking latest spark 2.3.2 and are unable to test running gpu workloads on soak environments, I would say on actual MWT day we should either 1) not run them or 2) deploy gpu dispatchers and run workload after the normal spark workloads are in good shape (eg 250-500 running jobs; no failures). If they run fine, then great! If not, shut em down immediately. WDYT?
cc @vishnu2kmohan

Reply:
Would it work if we used `mesosphere/spark:2.6.0-2.3.2-hadoop-2.7` instead?

By the way, Spark 2.3 is still pending: https://jira.mesosphere.com/browse/DCOS-49000

Reply:
Nah, the spark-dcos-gpu image actually includes installing all the cuda/cudnn libraries + Tensorflow in order to run [this TF app](https://github.com/mesosphere/spark-build/blob/master/scale-tests/batch_test.py#L57)

Reply:
I looked for a bit and couldn't find the Dockerfile for that image. Could you please share it? We should probably get it checked in this repo.

Reply:
ah sorry about that - this is the one: https://github.com/samvantran/spark-dcos-docker/blob/master/Dockerfile-statsD-metrics 
but it also builds from https://github.com/samvantran/spark-dcos-docker/blob/master/Dockerfile-minpatch-nv390-cuda9-cudnn7115-tf15

Reply:
Thanks Sam. I created https://jira.mesosphere.com/browse/DCOS-49094.

Reply:
I had a chat with SRE and they think they'll be able to add a GPU agent to the Soak cluster by tomorrow, so assuming we use the latest Spark version, we'll be able to test whether or not using the "spark-dcos-gpu" (which lower Spark version) works for running jobs against it.

Reply:
Pushed commit taking advantage of the GPU node: https://github.com/mesosphere/spark-build/pull/487/commits/9b45961ebcf3c555334b248c46a2b9a27269b1e7

Reply:
@samvantran on today's test I noticed that GPU jobs were not getting scheduled by Spark: https://soak113s.testing.mesosphe.re/service/data-services-testmwst13/gpu-spark-00/

Maybe it's related to the current Docker image with CUDA libraries not being compatible with the newest Spark version? If you have some time, could you take a look into this @samvantran?

In any case I'll attempt to submit the GPU jobs tomorrow and see what happens. Unless you discover that the Docker image Spark version mismatch is indeed the reason.

Reply:
I don't think it's a matter of the docker image. These jobs just aren't being scheduled. 

My guess is that soak113 is not set up the same as MWST. From `/mesos/flags` I see 
```
"filter_gpu_resources":"true"
fair_sharing_excluded_resource_names":"{ gpus }"
```
which leads me to [MESOS-5377](https://issues.apache.org/jira/browse/MESOS-5377) which seems to affect DRF strategy specifically for what is happening in this cluster, 1 gpu agent vs many non-gpu. 

I would say still worth attempting to run GPU jobs tmrw and if they don't run we can triage why then.

Reply:
Ack. Thanks for taking a look!